Core/Python/update_firmware_using_catalog.py

```diff
@@ -144,7 +154,40 @@ def catalog_creation_payload(**kwargs):
         }
     }
 
-def catalog_creation(ip_address, headers, **kwargs):
+def catalog_refresh_payload(repoId):
+    refresh_payload = {
+            "CatalogIds": [repoId],
+            "AllCatalogs": "false"
+    }
+    return refresh_payload
+
+
+def create_or_refresh_catalog(ip_address, headers, **kwargs):
+    """Get all catalogs first"""
+    url = 'https://%s/api/UpdateService/Catalogs' % ip_address
+    status, data = request(url=url, header=headers, method='GET')
+    if status != 200:
+        raise Exception("Unable to get the catalog", data)
+    else:
+        allrepoProfiles  = data["value"]
+        if allrepoProfiles is not None and data["@odata.count"] != 0:
+            for repoProfile in allrepoProfiles:
+                repositoryType = repoProfile["Repository"]["RepositoryType"]
+                if(repositoryType == kwargs['repo_type']):
+                    if kwargs['refresh']:
+                        url = 'https://%s/api/UpdateService/Actions/UpdateService.RefreshCatalogs' % ip_address
+                        refresh_payload = catalog_refresh_payload(repoProfile["Id"])
+                        status, data = request(url=url,
+                                               header=headers,  payload=refresh_payload,method='POST')
+                        if status != 204:
+                            raise Exception("Unable to refresh the catalog of " + repositoryType, data)
+                        else:
+                            time.sleep(60)
+                            print("Catalog refreshed ")
```

Comment:
Is there a way to check if this completed / does this happen worst case within 60 seconds?

Reply:
The refresh job invocation appears to return a boolean and not a jobID.
But in several tests, the refresh completed within 60 seconds.